### PR TITLE
Add support for Docker Compose V2

### LINKED ISSUE #404: Naming convention is different between docker-compose v1 and v2
**Describe the bug**
Container naming convention is different between Docker Compose versions v1 (Python) and v2 (Go).

```python 
def rename_to_tmp_name(self):
        """Rename the container to a hopefully unique temporary container name
        by prepending the short id.
        """
        if not self.name.startswith(self.short_id):
            self.client.rename(
                self.id, '{}_{}'.format(self.short_id, self.name)
            )
```
https://github.com/docker/compose/blob/5db8d86f127848367e4d721e992c188923c89d42/compose/container.py#L260

```go
// Separator is used for naming components
var Separator = "-"
```
https://github.com/docker/compose/blob/0062703bea796eca5b5f78a4779fa166cc18619c/pkg/compose/compose.go#L36

Need to update hardcoded separator in containerName
https://github.com/testcontainers/testcontainers-go/blob/8be67da6e9e105251d7b73f0fdbacc1f9d6f0b28/compose.go#L147

**To Reproduce**

https://gist.github.com/karayel/07ecb1c1651bd7430e795ccf1631a8a8

**INFO: You can reproduce this bug with using docker-compose v2**

**Expected behavior**
Match running docker container with given containerName

** docker info **
output of the command:

```
$ docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.7.1)
  compose: Docker Compose (Docker Inc., v2.2.3)
  scan: Docker Scan (Docker Inc., v0.16.0)

Server:
 Containers: 21
  Running: 0
  Paused: 0
  Stopped: 21
 Images: 6
 Server Version: 20.10.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runtime.v1.linux runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.94GiB
 Name: docker-desktop
 ID: X2TY:Y4ES:JJR6:WOCD:5ZPL:ACVO:DLZ6:QN3E:VWKW:LICR:TPZ5:NGNR
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Additional context**

I am happy to contribution after discussing the solutions to the problem.

### PR DESCRIPTION
**Pull request by artamonovkirill**
* Fixes https://github.com/testcontainers/testcontainers-go/issues/374 
* Allows waiting for a container with `container_name` specified in the `docker-compose.yml`

## Related issues
- Closes #404
- Closes #371

### CODE REVIEW

**Comment by portey** [comment]
Hi guys, 
Thanks for your effort! 

Do you have plans to merge this pr in order to unblock all users who use docker-compose V2?

File: compose.go [hunk 1]
```diff
@@ -113,11 +113,12 @@ func (dc *LocalDockerCompose) applyStrategyToRunningContainer() error {
 	}
 
 	for k := range dc.WaitStrategyMap {
-		// TODO: if the compose file includes a container_name, this won't work,
-		// as we cannot delegate to the names generated by Docker Compose
 		containerName := dc.Identifier + "_" + k.service
-		// Docker compose appends "_1" to every started service by default. Trimming to match running docker container
-		f := filters.NewArgs(filters.Arg("name", containerName))
+		composeV2ContainerName := strings.Replace(containerName, "_", "-", -1)
```
Comment (mdelapenya) [inline compose.go:117]:
```suggestion
		composeV2ContainerName := strings.ReplaceAll(containerName, "_", "-")
```

Comment (karayel) [reply compose.go:117]: If this line is blocker for merge, I can create another pull request with this change. Is it blocker ?

Comment (mdelapenya) [reply compose.go:117]: Not a blocker at all, as I could directly commit it to this PR, but wanted to give @artamonovkirill feedback and time to see it before committing it

**Comment by mdelapenya** [comment]
As discussed in #404, I think this change is required to support multiple versions of docker-compose.

Thanks for your help during the review

**Pull request merged**